components/script/dom/htmllabelelement.rs

```diff
@@ -40,6 +42,43 @@ impl HTMLLabelElementMethods for HTMLLabelElement {
     fn GetForm(&self) -> Option<Root<HTMLFormElement>> {
         self.form_owner()
     }
+
+    // https://html.spec.whatwg.org/multipage/#dom-label-htmlfor
+    make_getter!(HtmlFor, "for");
+
+    // https://html.spec.whatwg.org/multipage/#dom-label-htmlfor
+    make_setter!(SetHtmlFor, "for");
+
+    // https://html.spec.whatwg.org/multipage/#dom-label-control
+    fn GetControl(&self) -> Option<Root<HTMLElement>> {
+        let for_ = self.HtmlFor();
+
+        if for_.is_empty() {
+            return self.first_labelable_descendant();
+        }
+
+        let elem = match document_from_node(self).GetElementById(for_)
+                                                 .and_then(Root::downcast::<HTMLElement>) {
+            Some(elem) => elem,
```

Comment:
`.iter().filter(|elem| elem.is_labelable_element()).next()` would be a bit shorter... or maybe you have some other idea?  Anyway, the current version seems extremely verbose.